core/src/test/java/com/radixdlt/consensus/safety/SafetyStoreMigrationTest.java

```diff
@@ -0,0 +1,170 @@
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.consensus.safety;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.inject.*;
+import com.google.inject.Module;
+import com.radixdlt.consensus.bft.BFTValidatorId;
+import com.radixdlt.environment.NodeRustEnvironment;
+import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.monitoring.MetricsInitializer;
+import com.radixdlt.rev2.NodeRustEnvironmentBuilder;
+import com.radixdlt.rev2.modules.NodePersistenceModule;
+import com.radixdlt.rev2.modules.NodePersistenceModule.AddressBookPolicy;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.core.ClasspathScanningSerializationPolicy;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+import com.radixdlt.store.BerkeleyDbDefaults;
+import com.radixdlt.store.NodeStorageLocation;
+import com.radixdlt.utils.properties.RuntimeProperties;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.commons.cli.ParseException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SafetyStoreMigrationTest {
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private Module createCommonModule() {
```

Comment:
Similar to the other test - can we avoid adding new guice modules in tests? They create a maintenance burden

Reply:
Reworked